core/modules/wdb/ChunkResource.cc

```diff
@@ -235,8 +279,112 @@ class Backend {
+        LOGF_DEBUG("execLockSql %1%" % query);
+        sql::SqlErrorObject err;
+        if(!_sqlConn.runQuery(query, err)) {
+            LOGF_ERROR("Lock failed, exiting. query=%1%  err=%2%" % query % err.printErrMsg());
+            exit(EXIT_FAILURE);
+        }
+    }
+
+    /// Return the status of the lock on the in memory tables.
+    LockStatus _memLockStatus() {
+        std::string sql = "SELECT uid FROM " + _lockDbTbl + " WHERE keyId = 1;";
+        sql::SqlResults results;
+        sql::SqlErrorObject err;
+        if (!_sqlConn.runQuery(sql, results, err)) {
+            // Assuming UNLOCKED should be safe as either it must be LOCKED_OURS to continue
+            // or we are about to try to lock. Failure to lock will cause the program to exit.
+            std::ostringstream emsg;
+            emsg << "memLockStatus query failed, assuming UNLOCKED. " << sql << " err=" << err.printErrMsg();
+            LOGF_WARN("%1%" % emsg.str());
+            return UNLOCKED;
+        }
+        std::string uidStr;
+        if (!results.extractFirstValue(uidStr, err)) {
+            std::ostringstream emsg;
+            emsg << "memLockStatus unexpected results, assuming LOCKED_OTHER. err=" << err.printErrMsg();
+            LOGF_WARN("%1%" % emsg.str());
+            return LOCKED_OTHER;
+        }
+        int uid = atoi(uidStr.c_str());
+        if (uid != _uid) {
+            std::ostringstream emsg;
+            emsg << "memLockStatus LOCKED_OTHER wrong uid. Expected " << _uid << " got " << uid
+                << " err=" << err.printErrMsg();
+            LOGF_WARN("%1%" % emsg.str());
+            return LOCKED_OTHER;
+        }
+        return LOCKED_OURS;
+    }
+
+    /// Take the lock on the in memory tables, even if held by another.
+    // Terminate the program if we can't take the lock.
+    // This must be run before any other operations on in memory tables.
+    void _memLockLock() {
+        _lockDb = MEMLOCKDB;
+        _lockTbl = MEMLOCKTBL;
+        _lockDbTbl = _lockDb + "." + _lockTbl;
+        LockStatus mls = _memLockStatus();
+        if (mls != UNLOCKED) {
+            LOGF_WARN("Memory tables were not released cleanly! LockStatus=%1%" % strLockStatus(mls));
+        }
+
+        // Lock the memory tables.
+        std::string sql = "CREATE DATABASE IF NOT EXISTS " + _lockDb + ";";
+        sql += "CREATE TABLE IF NOT EXISTS " + _lockDbTbl + " ( keyId INT UNIQUE, uid INT ) ENGINE = MEMORY;";
+        _execLockSql(sql);
+        sql = "TRUNCATE TABLE " + _lockDbTbl;
+        _execLockSql(sql);
+        std::ostringstream insert;
+        insert << "INSERT INTO " << _lockDbTbl << " (keyId, uid) VALUES(1, " << _uid << " );";
+        _execLockSql(insert.str());
+
+        // Delete any old in memory databases. They could be empty or wrong.
+        // Empty tables would prevent new tables from being created.
+        std::ostringstream os;
+        os << "SHOW DATABASES LIKE '" << SUBCHUNKDB_PREFIX << "%';";
```

Comment:
`SUBCHUNKDB_PREFIX` contains underscore which is a wildcard character in mysql LIKE. IT can potentially match more than what you need (e.g. in addition to "Subchunks_123" it will also match "SubchunksDoNotDelete"). I'd add additional check in the for loop below to verify that name actually starts with `SUBCHUNKDB_PREFIX`.